great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -121,18 +120,27 @@ def _query_for_year_and_month(
     return {"year": year, "month": month}
 
 
+SQL_DATETIME_COUNTER = 0
+
+
 def _get_sql_datetime_range(
-    conn: sqlalchemy.engine.base.Connection, table_name: str, col_name: str
+    conn: sqlalchemy.engine.base.Connection, data_from: str, col_name: str
 ) -> DatetimeRange:
-    q = f"select min({col_name}), max({col_name}) from {table_name}"
+    global SQL_DATETIME_COUNTER
+    from_name = f"_gx_sql_datetime_range_data_{SQL_DATETIME_COUNTER}"
```

Comment:
I use the `SQL_DATETIME_COUNTER` to make sure every subquery alias is unique. I could use a generated uuid in the function itself. Let me know if you prefer this or a different strategy.

Reply:
@billdirks In my opinion, something that ends on `_{str(uuid.uuid4())[:8]}` might be better.  We have a function in `great_expectations/util.py` called `generate_temporary_table_name()`; one can create a better-named one here by calling `generate_temporary_table_name()` with `default_table_name_prefix=""` and leaving the `num_digits` same.  Thanks.